File: code_BE/lms/src/main/java/com/lms/lms/service/EnrollmentService.java
package com.lms.lms.service;

import com.lms.lms.dto.response.EnrollCheckResponse;
import com.lms.lms.dto.response.EnrollmentResponse;

import java.util.List;

public interface EnrollmentService {
    // FREE course: enroll immediately
    EnrollmentResponse enrollFree(String courseId, String studentId);

    // check if student is enrolled
    EnrollCheckResponse checkEnrollment(String courseId, String studentId);

    // list enrolled courses for student
    List<EnrollmentResponse> getMyEnrollments(String studentId);

    // internal: create enrollment after payment success
    EnrollmentResponse createEnrollment(String courseId, String studentId);
}

File: codeFE/src/services/enrollmentService.ts
import api from "@/lib/axios";
import type { ApiResponse } from "@/types";

export interface EnrollmentResponse {
    id: string;
    courseId: string;
    studentId: string;
    enrolledAt: string;
}

export interface EnrollCheckResponse {
    enrolled: boolean;
    courseId: string;
}

export interface PaymentResponse {
    paymentId: string;
    courseId: string;
    amount: number;
    status: "PENDING" | "SUCCESS" | "FAILED";
    paymentUrl: string;
}

export const enrollmentService = {
    // FREE course
    enrollFree: (courseId: string) =>
        api.post<ApiResponse<EnrollmentResponse>>(`/api/enrollments/${courseId}`)
            .then((r: { data: ApiResponse<EnrollmentResponse> }) => r.data.result),

    // check enrollment
    checkEnrollment: (courseId: string) =>
        api.get<ApiResponse<EnrollCheckResponse>>(`/api/enrollments/check?courseId=${courseId}`)
            .then((r: { data: ApiResponse<EnrollCheckResponse> }) => r.data.result),

    // list my enrollments
    getMyEnrollments: () =>
        api.get<ApiResponse<EnrollmentResponse[]>>("/api/enrollments/my")
            .then((r: { data: ApiResponse<EnrollmentResponse[]> }) => r.data.result),

    // PAID course: create payment → get VNPay URL
    createPayment: (courseId: string) =>
        api.post<ApiResponse<PaymentResponse>>(`/api/payments/create/${courseId}`)
            .then((r: { data: ApiResponse<PaymentResponse> }) => r.data.result),
};
